item.rb

```diff
@@ -38,9 +38,10 @@ def move_to_archive
   end
 
   def parse_date(date_str)
-    Date.parse(date_str)
+    Date.parse(date_str.to_s)
   rescue ArgumentError, TypeError
-    raise ArgumentError, 'Invalid publish date format'
+    raise ArgumentError, 'Invalid publish date format. Please use the format YYYY-MM-DD.'
   end
```

Comment:
The code captures both ArgumentError and TypeError. While it's possible to encounter a TypeError if date_str is not a string or to_s method fails, it might be more appropriate to handle only ArgumentError since Date.parse already handles string conversion internally. Handling TypeError might mask other potential issues.